### PR TITLE
Add local up kubeedge script

### LINKED ISSUE #1077: Add a script to run kubeedge on one VM
**What would you like to be added/modified**:
Add a script which run kubeedge on one VM, which can be used to develop test and e2e test in CI.
Xref: [the k8s local-up-cluster script](https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh)

The k8s cluster can started used the [kind](https://github.com/kubernetes-sigs/kind), the kubeedge components run in the VM as process 

**Why is this needed**:
Used  for development testing.

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**
 /kind feature

**What this PR does / why we need it**:
Run this script can setup a kubeedge cluster in one VM, used for develop and CI test.
Use kind to start a k8s cluster, run cloudcore and edgecore in VM.

**Which issue(s) this PR fixes**:

Fixes #1077

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/lib/install.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  fi
+}
+
+verify_docker_installed(){
+  # verify the docker installed
+  command -v docker >/dev/null || {
+    echo "must install the docker first"
+    exit 1
+  }
+}
```
Comment (sids-b) [inline hack/lib/install.sh]: please add a new line

File: hack/local-up-kubeedge.sh [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+To start using your kubeedge, you can run:
+
+  export PATH=$PATH:$GOPATH/bin
+  export KUBECONFIG="$(kind get kubeconfig-path --name="test")"
+  kubectl get nodes
+"
+
+if [[ "${ENABLE_DAEMON}" = false ]]; then
+  while true; do sleep 1; healthcheck; done
+fi
```
Comment (sids-b) [inline hack/local-up-kubeedge.sh]: please add a new line

File: hack/lib/install.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+# check if kind installed
+function check-kind {
+  echo "checking kind"
+  which kind >/dev/null 2>&1
+  if [[ $? -ne 0 ]]; then
+    echo "installing kind ."
+    GO111MODULE="on" go get sigs.k8s.io/kind@v0.5.1
+    export PATH=$PATH:$GOPATH/bin
```
Comment (sids-b) [inline hack/lib/install.sh]: I think this step should be one of the prerequisites. It may add GOPATH/bin to the PATH twice

Comment (kevin-wangzefeng) [reply hack/lib/install.sh]: It only try add $GOPAHT/bing to $PATH when kind is newly installed, doesn't really matter if added twice.

File: hack/local-up-kubeedge.sh [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+
+if [[ "${CLUSTER_NAME}x" == "x" ]];then
+    CLUSTER_NAME="test"
+fi
+
+export CLUSTER_CONTEXT="--name ${CLUSTER_NAME}"
+
+# spin up cluster with kind command
+function kind-up-cluster {
```
Comment (kevin-wangzefeng) [inline hack/local-up-kubeedge.sh]: make all function names with "_" instead of  "-".

File: hack/local-up-kubeedge.sh [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+  kubectl apply -f ${KUBEEDGE_ROOT}/build/crds/devices/devices_v1alpha1_devicemodel.yaml
+}
+
+function build-cloudcore {
+  # TODO: make the binary to _output dir
+  echo "building the cloudcore..."
+  make -C "${KUBEEDGE_ROOT}" WHAT="cloud"
+  mkdir -p ${KUBEEDGE_ROOT}/_output/bin/cloud
+  mv ${KUBEEDGE_ROOT}/cloud/cloudcore ${KUBEEDGE_ROOT}/_output/bin/cloud
+  cp -r ${KUBEEDGE_ROOT}/cloud/conf ${KUBEEDGE_ROOT}/_output/bin/cloud
```
Comment (kevin-wangzefeng) [inline hack/local-up-kubeedge.sh]: TODO: let the make generate example yaml file under target output folder.

File: hack/local-up-kubeedge.sh (hunk 4 above)
Comment (fisherxu) [reply hack/local-up-kubeedge.sh]: done.

File: hack/local-up-kubeedge.sh (hunk 5 above)
Comment (fisherxu) [reply hack/local-up-kubeedge.sh]: sure, will do later.

**Comment by fisherxu** [comment]
@kevin-wangzefeng have done,ptal :)

**Comment by fisherxu** [comment]
/hold 
wait #1090 merged first, need rebased

**Pull request merged**